### PR TITLE
TENTACLES-12: Update to latest log4j2 and update dependencies and plugins

### PR DESCRIPTION
**Pull request by ottlinger**
https://issues.apache.org/jira/projects/TENTACLES/issues/TENTACLES-12

As log4j2 2.14.0 is built with JDK9 minimum some more updates are needed apart from build configuration and compilation changes in order to use log2j2 instead of v1.x with security problems.

### CODE REVIEW

**Comment by ottlinger** [comment]
@jvz I tried to get this animal-sniffer-stuff up and running with jdk9 but it complained that log4j is JDK9.
As sniffer only supports JDK8 I had to remove it.
Any improvents are welcome if you have a running animal-sniffer-verification setup.
Thanks

**Comment by jvz** [comment]
https://github.com/mojohaus/animal-sniffer/pull/32

Seems like it was fixed a while ago (and by coworkers of mine, too, amusingly enough).

**Comment by ottlinger** [comment]
@jvz we ran with v1.16 - the problem is that there is no

```
<groupId>org.codehaus.mojo.signature</groupId>
<artifactId>java19</artifactId>
```
in order to scan for at least JDK9.

Runnig with java18 yielded the problem with log4j.

**Comment by jvz** [comment]
Here's a more thorough example of what we've done in Jenkins to work around that: https://github.com/jenkinsci/plugin-pom/blob/3101645162a2c0a210fbd470140b7013fcd9799f/pom.xml#L464-L549

**Comment by rmannibucau** [comment]
Guess we can also simply move creadur to JUL, there is no real reason to use log4j (in particular until it gets v3 cause it is way too fat and slow to start these days :().

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: java
 jdk:
-  - openjdk8
+  - openjdk9
```
Comment (rmannibucau) [inline .travis.yml:3]: Guess we must keep java 8 there.
Between that and disabling  a buggy plugin (animal sniffer) guess last one is saner.

**Comment by ottlinger** [comment]
@rmannibucau the Travis build is buggy as you get a different JDK than the one you selected as build JDK. This yielded various build failures in the past and is a known-issue at Travis.
The failed builds ran with JDK11 although 9 was requested ..... this makes builds fail that use animal-sniffer stuff, the build is/should be green on ASF infra.

**Comment by ottlinger** [comment]
After some changes in the enforcer-plugin configuration the build runs successfully with JDK8 - thanks for all your input.

**Pull request merged**